plugins/rethinkdb/rethinkdb.go

```diff
@@ -0,0 +1,330 @@
+// Returns one of the errors encountered while gather stats (if any).
+func (r *RethinkDB) Gather(acc plugins.Accumulator) error {
+	if len(r.Servers) == 0 {
+		url := &url.URL{
+			Host: "127.0.0.1:28015",
+		}
+		r.gatherServer(url, acc)
+		return nil
+	}
+
+	var wg sync.WaitGroup
+
+	var outerr error
+
+	for _, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("Unable to parse to address '%s': %s", serv, err)
+		} else if u.Scheme == "" {
+			// fallback to simple string based address (i.e. "10.0.0.1:10000")
+			u.Host = serv
+		}
+		wg.Add(1)
+		go func(serv string) {
+			defer wg.Done()
+			outerr = r.gatherServer(u, acc)
+		}(serv)
+	}
+
+	wg.Wait()
+
+	return outerr
+}
+
+type serverStatus struct {
+	Id      string `gorethink:"id"`
+	Network struct {
+		Addresses  []Address `gorethink:"canonical_addresses"`
+		Hostname   string    `gorethink:"hostname"`
+		DriverPort int       `gorethink:"reql_port"`
+	} `gorethink:"network"`
+	Process struct {
+		Version      string    `gorethink:"version"`
+		RunningSince time.Time `gorethink:"time_started"`
+	} `gorethink:"process"`
+}
+
+type Address struct {
+	Host string `gorethink:"host"`
+	Port int    `gorethink:"port"`
+}
+
+type stats struct {
+	Engine engine `gorethink:"query_engine"`
+}
+
+type engine struct {
+	ClientConns   int `gorethink:"client_connections"`
+	ClientActive  int `gorethink:"clients_active"`
+	QueriesPerSec int `gorethink:"queries_per_sec"`
+	TotalQueries  int `gorethink:"queries_total"`
+	ReadsPerSec   int `gorethink:"read_docs_per_sec"`
+	TotalReads    int `gorethink:"read_docs_total"`
+	WritesPerSec  int `gorethink:"written_docs_per_sec"`
+	TotalWrites   int `gorethink:"written_docs_total"`
+}
+
+type tableStatus struct {
+	Id   string `gorethink:"id"`
+	DB   string `gorethink:"db"`
+	Name string `gorethink:"name"`
+}
+
+type tableStats struct {
+	Engine  engine `gorethink:"query_engine"`
+	Storage struct {
+		Cache struct {
+			BytesInUse int `gorethink:"in_use_bytes"`
+		} `gorethink:"cache"`
+		Disk struct {
+			ReadBytesPerSec  int `gorethink:"read_bytes_per_sec"`
+			ReadBytesTotal   int `gorethink:"read_bytes_total"`
+			WriteBytesPerSec int `gorethik:"written_bytes_per_sec"`
+			WriteBytesTotal  int `gorethink:"written_bytes_total"`
+			SpaceUsage       struct {
+				Data     int `gorethink:"data_bytes"`
+				Garbage  int `gorethink:"garbage_bytes"`
+				Metadata int `gorethink:"metadata_bytes"`
+				Prealloc int `gorethink:"preallocated_bytes"`
+			} `gorethink:"space_usage"`
+		} `gorethink:"disk"`
+	} `gorethink:"storage_engine"`
+}
+
+func (r *RethinkDB) gatherServer(serv *url.URL, acc plugins.Accumulator) error {
+	var err error
+	connectOpts := gorethink.ConnectOpts{
+		Address:       serv.Host,
+		DiscoverHosts: false,
+	}
+	if serv.User != nil {
+		pwd, set := serv.User.Password()
+		if set && pwd != "" {
+			connectOpts.AuthKey = pwd
+		}
+	}
+	r.session, err = gorethink.Connect(connectOpts)
```

Comment:
You don't close `r.session` here (or close a previous one) so you'll open infinite new connections.